grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -402,27 +402,31 @@ class UserService {
         return false
     }
 
-    def getDashboardConfig(User user) {
-        def config = grailsApplication.config.openboxes.dashboardConfig
+    def getDashboardConfig(User user, String id) {
+        def fullConfig = grailsApplication.config.openboxes.dashboardConfig
+        def resultConfig = [
+                dashboards: fullConfig.dashboards[id ?: "mainDashboard"],
+                dashboardWidgets: fullConfig.dashboardWidgets
+        ]
         def userConfig = user.deserializeDashboardConfig()
         Boolean configChanged = false
 
         if (userConfig != null) {
             int userConfigSize = userConfig.size()
-            int configSize = config.dashboards.size()
+            int resultConfigSize = resultConfig.dashboards.size()
             // If the size is different, that mean that the config has changed
-            if (userConfigSize != configSize) {
-                return config
+            if (userConfigSize != resultConfigSize) {
```

Comment:
If the userConfig (which is based on the main dashboard) will have the same size as for example supplier dashboard stored in the resultConfig, then the user config will be returned instead of another dashboard by id.